From 81bc0cdc0e9baa192326e5dc2f682974c45f43f8 Mon Sep 17 00:00:00 2001
From: Xiaowen Liu <b37945@freescale.com>
Date: Thu, 25 Jul 2013 14:55:19 +0800
Subject: [PATCH 460/635] ENGR00272567 add BGRA8888 support in hwcomposer and
 pass to framework.

add BGRA8888 support in hwcomposer/gralloc and pass to framework.

Signed-off-by: Xiaowen Liu <b37945@freescale.com>
---
 mx6/hwcomposer/hwc_display.cpp         | 3 ++-
 mx6/libgralloc_wrapper/framebuffer.cpp | 7 ++++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/mx6/hwcomposer/hwc_display.cpp b/mx6/hwcomposer/hwc_display.cpp
index f4ba655..d34def4 100644
--- a/mx6/hwcomposer/hwc_display.cpp
+++ b/mx6/hwcomposer/hwc_display.cpp
@@ -190,7 +190,8 @@ int hwc_get_framebuffer_info(displayInfo *pInfo)
     pInfo->vsync_period  = 1000000000 / refreshRate;
     pInfo->blank  = 0;
     pInfo->format = (info.bits_per_pixel == 32)
-                         ? HAL_PIXEL_FORMAT_RGBA_8888
+                         ? ((info.red.offset == 0) ? HAL_PIXEL_FORMAT_RGBA_8888 :
+                            HAL_PIXEL_FORMAT_BGRA_8888)
                          : HAL_PIXEL_FORMAT_RGB_565;
 
     ALOGV("using\n"
diff --git a/mx6/libgralloc_wrapper/framebuffer.cpp b/mx6/libgralloc_wrapper/framebuffer.cpp
index 6f62cd6..18971eb 100755
--- a/mx6/libgralloc_wrapper/framebuffer.cpp
+++ b/mx6/libgralloc_wrapper/framebuffer.cpp
@@ -450,7 +450,12 @@ static void fb_device_init(private_module_t* m, fb_context_t *dev)
 	const_cast<int&>(dev->device.format) = HAL_PIXEL_FORMAT_RGB_565;
     }
     else{
-	const_cast<int&>(dev->device.format) = HAL_PIXEL_FORMAT_RGBA_8888;
+        if (m->info.red.offset == 0) {
+	    const_cast<int&>(dev->device.format) = HAL_PIXEL_FORMAT_RGBA_8888;
+        }
+        else {
+	    const_cast<int&>(dev->device.format) = HAL_PIXEL_FORMAT_BGRA_8888;
+        }
     }
     const_cast<float&>(dev->device.xdpi) = m->xdpi;
     const_cast<float&>(dev->device.ydpi) = m->ydpi;
-- 
1.8.0

